setup.py

```diff
@@ -37,20 +37,20 @@
 """
 
 import glob
-import os.path
+import os
 import subprocess
 import sys
 
 from distutils.spawn import find_executable
 from pkg_resources import get_build_platform
-from setuptools import setup, find_packages
+from setuptools import setup
 from setuptools.extension import Extension
 
 # Execute DAKOTA to get version.
 try:
```

Comment:
These `try: except:` blocks which just immediately rethrow with a nicer message or call `sys.exit( )` do in my opinion only clutter the code. As long we do not intend to really handle the exception - or let it affect program flow, we should let it bubble up, i.e. I would just do (ignore the formatting):
```python
dakota_process = subprocess.Popen(['dakota', '-v'],  universal_newlines=True,
                                                     stdout=subprocess.PIPE, 
                                                     stderr=subprocess.PIPE)                                 
stdout, stderr = dakota_process.communicate()
```
without *any* `try: except:`

In general I think these are good rules for use of exceptions:

1. The `try` and the `except` should be **very** close together in the code; only one logical "thing" should be perfomed in the `try:` block - and ideally that *thing* should consist of only one statement.
2. Avoid the use catch all `except:`  statements - the catch statement should very specifically look for the one single (or very very few) exception types which you have analyzed can happen - other exceptions you are not really prepared to handle, and they should just bubble up.

Others might disagree though.